Core/src/org/sleuthkit/autopsy/datamodel/InterestingHits.java

```diff
@@ -345,23 +347,28 @@ private HitFactory(String setName) {
 
         @Override
         protected boolean createKeys(List<Long> list) {
-            for (long l : interestingResults.getArtifactIds(setName)) {
-                list.add(l);
+            
+            if (skCase == null) {
+               return true;
             }
+            
+            interestingResults.getArtifactIds(setName).forEach((id) -> {
+                try {
+                    BlackboardArtifact art = skCase.getBlackboardArtifact(id);
+                    
+                    artifactHits.put(id, art);
+                    list.add(id);
+                } catch (TskCoreException ex) {
```

Comment:
Since this is a case database error, will you please change the log level to SEVERE instead of WARNING, while we are in this code?